server/src/main/java/org/elasticsearch/search/retriever/KnnRetrieverBuilder.java

```diff
@@ -217,7 +217,8 @@ public QueryBuilder explainQuery() {
         var rankDocsQuery = new RankDocsQueryBuilder(
             rankDocs,
             new QueryBuilder[] { new ExactKnnQueryBuilder(VectorData.fromFloats(queryVector.get()), field, similarity) },
-            true
+            false,
+            Float.MIN_VALUE
```

Comment:
This should use a default member that you create and reference as suggested earlier. Alternately consider a convenience constructor that sets the default.